vm_supervisor/pubsub.py

```diff
@@ -45,12 +45,11 @@ async def msubscribe(self, *keys):
 
         # Cleanup: remove the queue from the subscribers
         for key in keys:
-            for subscriber in self.subscribers.values():
+            for subscriber in list(self.subscribers.values()):
                 subscriber.discard(queue)
                 # Remove keys with no remaining queue
-                if not self.subscribers.get(key):
+                if self.subscribers.get(key) == set():
```

Comment:
Mmh this could be dangerous if we ever use a different type than set. Maybe just check `if not self.subscribers:` ?

Reply:
As discussed, I did this instead (and an additional comment):

![image](https://user-images.githubusercontent.com/404665/189922663-dbbbffe6-e00f-494e-84dc-4365bc9b2c9c.png)